include/fmt/format.h

```diff
@@ -3978,9 +3978,16 @@ struct formatter<T, Char, enable_if_t<detail::has_format_as<T>::value>>
   }
 };
 
-#define FMT_FORMAT_AS(Type, Base) \
-  template <typename Char>        \
-  struct formatter<Type, Char> : formatter<Base, Char> {}
+#define FMT_FORMAT_AS(Type, Base)                                   \
+  template <typename Char>                                          \
+  struct formatter<Type, Char> : formatter<Base, Char> {            \
+    template <typename FormatContext>                               \
+    auto format(Type value, FormatContext&& ctx) const              \
```

Comment:
We either have to pass `Type` as a value like here, or turn this parameter into a forwarding reference too otherwise we get const errors. I opted for a value as the types for this are all small (<=16bytes)

Ditto on const issues with passing `const format_context&` or `format_context&` so have used a forwarding ref for that too (if there's a better way for this one let me know)

Reply:
`FormatContext` should always be passed by a normal reference (`&`).